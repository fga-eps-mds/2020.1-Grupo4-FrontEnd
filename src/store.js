import { configureStore } from '@reduxjs/toolkit';
import usersReducer from './slices/usersSlice';
import tutorialSlice from './slices/tutorialSlice'

export default configureStore({
  reducer: {
    users: usersReducer,
    tutorial: tutorialSlice
  }
});