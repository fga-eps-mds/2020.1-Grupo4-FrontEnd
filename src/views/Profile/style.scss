.profile {
  min-height: 100vh;
  max-width: 80%;
  margin: 0 auto;
  padding: 5rem 1rem;
  font-family: Overpass;

  @media(max-width: 600px) {
    max-width: 90%;
    flex-direction: column;
    padding: 5rem 0;
  }

  &__header {
    margin-bottom: 2rem;
    h5 {
      padding-top: 0.2rem;
      text-align: center;
      width: 7rem;
      height: 1.8rem;
      color: var(--secondary-color);
      border-style: solid;
      border-radius: 20px;
      border-width: thin;
      border-color: var(--secondary-color);
    }
  }

  &__content {
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 5px 10px rgba(43, 43, 43, 0.07), 0px 15px 40px rgba(0, 0, 0, 0.04);
    margin-bottom: 2rem;
    padding: 2rem 3rem;

    @media(max-width: 600px) {
      margin-top: 2rem;
      flex-direction: column;
      padding: 1rem;
    }

    &--form {
      form {
        display: flex;
        flex-wrap: wrap;
        justify-content: space-between;

        @media(max-width: 600px) {
          flex-direction: column;
        }
      }

      .custom-input {
        width: 45%;

        @media(max-width: 600px) {
          width: 100%;
        }
      }
    }

    .form-action {
      display:  flex;
      align-items: center;
      flex-direction: column;
      padding: 1rem 0;
      margin: 0 auto;
      text-align: center;

      @media(max-width: 600px) {
        width: 100%;
      }

      button {
        width: 15rem;
        margin-bottom: 1.5rem;
      }

      p {
        font-weight: 600;
      }
    }
  }
}
