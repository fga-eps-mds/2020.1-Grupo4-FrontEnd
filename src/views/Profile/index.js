import React from 'react';
import PropTypes from 'prop-types'
import { useForm } from 'react-hook-form';
import { connect } from 'react-redux'
import { selectCurrentUser } from '../../slices/usersSlice';
import { editUser } from '../../services/usersService';
import avatar  from '../../assets/images/avatar.svg'
import Button from '../../components/Button';
import '../../index.css'
import './style.scss';

/* eslint-disable no-shadow */
/* eslint-disable jsx-a11y/label-has-associated-control */
function Profile({ currentUser, editUser }) {
    const { handleSubmit, register, errors } = useForm();

    const onSubmit = data => {
        editUser(data)
    }

    return (
        <>
            <div id="profile">
                <div className="profile__form">
                    <div className="profile__form--infos">
                        <img src={currentUser.profileImg || avatar} alt="" />
                        <div>
                            <h1>Perfil</h1>
                            <p>Adicione ou edite suas informações</p>
                        </div>
                        <Button>{`${currentUser.userType.toUpperCase()}${currentUser.isValidated ? ' EM ESPERA':''}`}</Button>
                    </div>
                    <form onSubmit={handleSubmit(onSubmit)}>
                        <div className="profile__inputs">
                            <label htmlFor="name">nome</label>
                                <input
                                    type="text"
                                    name="name"
                                    defaultValue={currentUser.name}
                                    ref={register({
                                        required: "campo obrigatório",
                                        pattern: {
                                            value: /^[A-Za-z][A-Za-z\s]*$/,
                                            message: "Nome inválido"
                                        }
                                    })} />

                                {(errors.name && <span className="danger">{errors.name.message}</span>) || <br />}
                
                            <label htmlFor="email">email</label>
                                <p>email</p> <input
                                    type="text"
                                    name="email"
                                    placeholder="email@email.com"
                                    defaultValue={currentUser.email}
                                    ref={register({
                                        required: "campo obrigatório",
                                        pattern: {
                                            value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                                            message: "email inválido"
                                        }
                                    })}
                                />
                                {(errors.email && <span className="danger">{errors.email.message}</span>) || <br />}

                            <label htmlFor="about">sobre você</label>
                                <p>sobre você</p> <textarea
                                    type="textarea"
                                    name="about"
                                    placeholder="Digite aqui uma mensagem sobre vocẽ"
                                    defaultValue={currentUser.about}
                                    ref={register({
                                        maxLength: {
                                            value: 300,
                                            message: "máximo de caracteres: 300"
                                        }
                                    })}
                                />
                                {(errors.about && <span className="danger">{errors.about.message}</span>) || <br />}

                            <label htmlFor="profileImg">
                                <span>Link da sua foto</span> <small>(comece com https://)</small>
                                <input
                                    name="profileImg"
                                    defaultValue={currentUser.profileImg}
                                    type="url"
                                    ref={register} />
                            </label>
                        </div>
                        <Button type="submit">ATUALIZAR</Button>
                    </form>
                </div>
            </div>
        </>
    );
}

Profile.propTypes = {
    currentUser: PropTypes.oneOfType([
        PropTypes.oneOf([null]),
        PropTypes.object
      ]).isRequired,
      editUser: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
    currentUser: selectCurrentUser(state)
})

const mapDispatchToProps = dispatch => ({
    editUser: (data) => dispatch(editUser(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(Profile)