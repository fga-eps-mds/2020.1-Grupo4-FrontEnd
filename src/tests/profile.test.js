import { shallow, mountW } from 'enzyme';
import React from 'react';
import Profile from '../views/Profile';
import 'mutationobserver-shim';
import { store } from '../store'

global.MutationObserver = window.MutationObserver;

it('Expect to render Login Component', () => {
  expect(shallow(<Profile store={store}></Profile>).length).toEqual(1)
  expect(shallow(<Profile store={store}></Profile>)).toMatchSnapshot()
})

it('Expect to click button', () => {
    const wrapper = shallow(<Profile store={store} />);
    wrapper.dive('Button').simulate('click')
  })

