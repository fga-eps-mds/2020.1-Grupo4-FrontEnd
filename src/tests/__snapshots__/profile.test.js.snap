// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Profile Component 1`] = `
<ContextProvider
  value={null}
>
  <Profile
    currentUser={null}
    editUser={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
